skbio/metadata/tests/test_interval.py

```diff
@@ -0,0 +1,505 @@
+
+    def test_set_boundaries_bad(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     locations=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        for value in [[(False, False)], ()]:
+            with self.assertRaises(ValueError):
+                f.boundaries = value
+        for value in [1, True]:
+            with self.assertRaises(TypeError):
+                f.boundaries = value
+
+    def test_get_metadata(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     locations=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        f.metadata['name'] = 'sagB'
+        self.assertEqual(f.metadata, {'name': 'sagB', 'function': 'transport'})
+
+    def test_set_metadata(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     locations=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        f.metadata = {'name': 'sagB', 'function': 'transport'}
+        self.assertDictEqual(f.metadata,
+                             {'name': 'sagB', 'function': 'transport'})
+        f.metadata = {}
+        self.assertDictEqual(f.metadata, {})
+
+    def test_set_metadata_bad(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     locations=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        for value in [1, '', None]:
+            with self.assertRaises(TypeError):
+                f.metadata = value
+
+    def test_set_get_bad(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     locations=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        f.drop()
+        with self.assertRaises(RuntimeError):
+            f.boundaries
+        with self.assertRaises(RuntimeError):
+            f.locations
+        with self.assertRaises(RuntimeError):
+            f.metadata
+        with self.assertRaises(RuntimeError):
+            f.boundaries = None
+        with self.assertRaises(RuntimeError):
+            f.locations = [(1, 2)]
+        with self.assertRaises(RuntimeError):
+            f.metadata = {}
+
+
+class TestIntervalUtil(unittest.TestCase):
+    def test_assert_valid_location(self):
+        intvls = [(1, 2), (-1, 2)]
+        for intvl in intvls:
+            try:
+                _assert_valid_location(intvl)
+            except:
```

Comment:
same comment re: not using blanket `except` statements, use explicit check or a more specific `except` statement